libpromises/evalfunction.c

```diff
@@ -7097,6 +7097,17 @@ static FnCallResult ValidateDataGeneric(const char *const fname,
         Log(LOG_LEVEL_VERBOSE, "%s: %s", fname, JsonParseErrorToString(err));
     }
 
+    for (const char *ch = data + 1; *ch != '\0'; ch++)
```

Comment:
So `JsonParse` advances the pointer to the last byte it processed? I'd add a comment about it.